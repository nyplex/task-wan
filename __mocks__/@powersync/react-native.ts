module.exports = {
  PowerSyncDatabase: jest.fn().mockImplementation(() => ({
    connect: jest.fn(),
    ready: true,
  })),
};
