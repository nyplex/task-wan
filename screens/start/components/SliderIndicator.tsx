import { Box } from "@/components/gluestack/box";
import clsx from "clsx";

type Props = {
  length: number;
  currentIndex?: number;
};

const SliderIndicator = ({ length, currentIndex }: Props) => {
  const getIndicatorStyle = (index: number) => {
    return clsx(
      "w-[10px] h-[10px] rounded-full mx-1",
      index === currentIndex ? "bg-primary-100" : "bg-primary-700"
    );
  };

  return (
    <Box className="flex-row items-center mt-4 ">
      {Array.from({ length }, (_, index) => (
        <Box
          key={index}
          className={getIndicatorStyle(index)}
        />
      ))}
    </Box>
  );
};

export default SliderIndicator;
