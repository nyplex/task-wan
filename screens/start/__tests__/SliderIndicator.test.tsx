import { render } from "@testing-library/react-native";
import SliderIndicator from "../components/SliderIndicator";
import React from "react";

describe("SliderIndicator", () => {
  it("renders the correct number of indicators", () => {
    const { getAllByTestId } = render(
      <SliderIndicator
        length={5}
        currentIndex={2}
      />
    );
    expect(getAllByTestId(/slider-indicator-/).length).toBe(5);
  });

  it("applies the correct style to the active indicator (snapshot)", () => {
    const { toJSON } = render(
      <SliderIndicator
        length={3}
        currentIndex={1}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders with no active indicator if currentIndex is undefined (snapshot)", () => {
    const { toJSON } = render(<SliderIndicator length={3} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders with length=1 (snapshot)", () => {
    const { toJSON } = render(
      <SliderIndicator
        length={1}
        currentIndex={0}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
