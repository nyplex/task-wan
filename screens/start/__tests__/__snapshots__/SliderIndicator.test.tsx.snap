// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderIndicator applies the correct style to the active indicator (snapshot) 1`] = `
<View
  className="flex-row items-center mt-4"
>
  <View
    className="w-[10px] h-[10px] rounded-full mx-1 bg-primary-700"
    testID="slider-indicator-0"
  />
  <View
    className="w-[10px] h-[10px] rounded-full mx-1 bg-primary-100"
    testID="slider-indicator-1"
  />
  <View
    className="w-[10px] h-[10px] rounded-full mx-1 bg-primary-700"
    testID="slider-indicator-2"
  />
</View>
`;

exports[`SliderIndicator renders with length=1 (snapshot) 1`] = `
<View
  className="flex-row items-center mt-4"
>
  <View
    className="w-[10px] h-[10px] rounded-full mx-1 bg-primary-100"
    testID="slider-indicator-0"
  />
</View>
`;

exports[`SliderIndicator renders with no active indicator if currentIndex is undefined (snapshot) 1`] = `
<View
  className="flex-row items-center mt-4"
>
  <View
    className="w-[10px] h-[10px] rounded-full mx-1 bg-primary-700"
    testID="slider-indicator-0"
  />
  <View
    className="w-[10px] h-[10px] rounded-full mx-1 bg-primary-700"
    testID="slider-indicator-1"
  />
  <View
    className="w-[10px] h-[10px] rounded-full mx-1 bg-primary-700"
    testID="slider-indicator-2"
  />
</View>
`;
